main.go

```diff
@@ -624,8 +626,10 @@ func main() {
 	}
 
 	// Cache swift PM
-	if err := cache.CollectPackagesCache(absProjectPath); err != nil {
-		log.Warnf("Failed to mark swift packages for caching, error: %s", err)
+	if configs.CacheLevel == "swift_packages" {
```

Comment:
pls move this block after the last os.exit call (do not cache if something still can break)